app/logic/Contacts/ActiveSync/ActiveSyncPerson.ts

```diff
@@ -1,4 +1,123 @@
-import { Person } from '../../Abstract/Person';
+import { Person, ContactEntry } from '../../Abstract/Person';
+import type { ActiveSyncAddressbook } from './ActiveSyncAddressbook';
+import { SQLPerson } from '../SQL/SQLPerson';
+import { EASError } from "../../Mail/ActiveSync/ActiveSyncAccount";
+import { assert } from "../../util/util";
+import { sanitize } from "../../../../lib/util/sanitizeDatatypes";
+import { parseOneAddress, type ParsedMailbox } from "email-addresses";
+
+const PhysicalAddressElements = ["Street", "City", "PostalCode", "State", "Country"];
+const PhoneMapping: [string, string, number][] = [
+  ["home", "tel", 2],
+  ["work", "tel", 2],
+  ["home", "fax", 1],
+  ["work", "fax", 1],
+  ["mobile", "tel", 1],
+];
+
+enum ContactElements {
+  home = "Home",
+  work = "Business",
+  other = "Other",
+  mobile = "Mobile",
+  tel = "Phone",
+  fax = "Fax",
+};
 
 export class ActiveSyncPerson extends Person {
+  addressbook: ActiveSyncAddressbook | null;
+
+  get serverID() {
+    return this.id;
+  }
+  set serverID(val) {
+    this.id = val;
+  }
+
+  fromWBXML(wbxmljs: any) {
+    this.firstName = sanitize.nonemptystring(wbxmljs.FirstName, "");
+    this.lastName = sanitize.nonemptystring(wbxmljs.LastName, "");
+    this.name = this.firstName ? this.lastName ? `${this.firstName} ${this.lastName}` : this.firstName : this.lastName;
+    this.emailAddresses.replaceAll([wbxmljs.Email1Address, wbxmljs.Email2Address, wbxmljs.Email3Address].filter(Boolean).map(address => new ContactEntry((parseOneAddress(address) as ParsedMailbox).address, "work", "mailto")));
```

Comment:
Can you please document what format `wbxmljs.Email1Address` is in? I'm assuming there's a reason why you use the `email-addresses` module. Is there a "Bob Sample <bob@example.com>" name in there? Or is just just about the `<>`s?

Could you please wrap the result of `parseOneAddress().address` with `sanitize.emailAddress()` ?

Reply:
Yes, it's a display address with `<>`s, which  GitHub is inconveniently hiding for us.

Sure, I can wrap a complex address parser with a simple sanitiser, but you won't get any extra protection.